java/openjdk-java/Portfile

```diff
@@ -0,0 +1,302 @@
+    version             15.0.7+1
+    set bv 15.0.7
+    set tpath /Library/Java
+    description         Openjdk 15 updates
+    long_description    Openjdk 15 updates of Openjdk, Open-Source implementation of the Java Platform, Standard Edition, and related projects.
+    homepage            https://openjdk.java.net
+    supported_archs     x86_64
+    fetch.type          git
+    git.url             https://git.openjdk.java.net/jdk15u
+    use_xcode           yes
+    use_configure    yes
+    configure.cmd       sh configure
+    configure.pre_args  --prefix=${tpath}
+    configure.args      --with-boot-jdk=/Library/Java/JavaVirtualMachines/jdk-15.0.2.jdk/Contents/Home \
+                        --with-conf-name=openjdk${bv}
+    build.type          gnu
+    build.target        images
+    build.post_args     CONF_CHECK=ignore
+    use_parallel_build  no
+    depends_build       port:autoconf \
+                        port:gmake
+    depends_lib         port:openjdk15-boot
+    set jdkn jdk-${bv}.jdk
+    set bundle_dir /build/openjdk${bv}/images/jdk-bundle/${jdkn}
+}
+
+subport openjdk17u {
+    # https://github.com/openjdk/jdk17u/tags
+    version             17.0.3+4
+    set bv 17.0.3
+    set tpath /Library/Java
+    description         Openjdk 17 updates
+    long_description    Openjdk 17 updates of Openjdk, Open-Source implementation of the Java Platform, Standard Edition, and related projects.
+    homepage            https://openjdk.java.net
+    supported_archs     x86_64 arm64
+    fetch.type          git
+    git.url             https://git.openjdk.java.net/jdk17u
+    use_xcode           yes
+    use_configure    yes
+    configure.cmd       sh configure
+    configure.pre_args  --prefix=${tpath}
+    configure.args      --with-boot-jdk=/Library/Java/JavaVirtualMachines/openjdk17-oracle/Contents/Home \
+                        --with-conf-name=openjdk${bv}
+    build.type          gnu
+    build.target        images
+    build.post_args     CONF_CHECK=ignore
+    use_parallel_build  no
+    depends_build       port:autoconf \
+                        port:gmake
+    depends_lib         port:openjdk17-oracle
+    set jdkn jdk-${bv}.jdk
+    set bundle_dir /build/openjdk${bv}/images/jdk-bundle/${jdkn}
+}
+
+if {[string match *u ${subport}]} {
+    set pathb ${tpath}/JavaVirtualMachines
+
+    destroot {
+        xinstall -m 755 -d ${destroot}${pathb}
+        copy ${worksrcpath}/${bundle_dir} ${destroot}${pathb}
+    }
+
+    notes "
+    If you want to make ${subport} the default JDK, add this to shell profile:
+    
+        export JAVA_HOME=${pathb}/${jdkn}/Contents/Home
+    "
+} elseif {[string match *-boot ${subport}]} {
+    use_xcode           no
+    use_configure    no
+    build {}
+    destroot.violate_mtree      yes
+
+    set path /Library/Java/JavaVirtualMachines
+
+    destroot {
+        xinstall -m 755 -d ${destroot}${path}
+        copy ${worksrcpath} ${destroot}${path}
```

Comment:
A port should not ever be installing anything outside of the MacPorts prefix.  Please change it so that it installs within the MacPorts tree.

Reply:
I can't actually change that, it is the default path where the file needs to be installed.

Reply:
If the file default installation path was `/usr/local`, I would have changed it to `${prefix}` because /opt/local is a system directory like /usr/local but this files installation directory is `/Library/Java/JavaVirtualMachine`